src/test/java/pl/marcin/project/repository/PurchaseRepositoryListBasedTest.java

```diff
@@ -0,0 +1,154 @@
+package pl.marcin.project.repository;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import pl.marcin.project.model.Cup;
+import pl.marcin.project.model.Customer;
+import pl.marcin.project.model.Purchase;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class PurchaseRepositoryListBasedTest {
+    PurchaseRepository purchaseRepository = new PurchaseRepositoryListBased();
+
+    @Test
+    public void shouldReturnPurchaseById() {
+        //given
+        int id = 1;
+        Purchase expectedPurchase = Purchase.builder().id(id).build();
+        purchaseRepository.savePurchase(expectedPurchase);
+
+        //when
+        Purchase purchase = purchaseRepository.findPurchase(id);
+
+        //then
+        Assertions.assertEquals(id, purchase.getId());
+    }
+
+    @Test
+    public void shouldReturnPurchaseWithId2() {
```

Comment:
we don't need this test, this logic was tested in `shouldReturnPurchaseById`

Reply:
ok